pkg/provisioningv2/rke2/planner/initnode.go

```diff
@@ -39,19 +39,19 @@ func (p *Planner) setInitNodeMark(entry *planEntry) error {
 
 // findAndDesignateFixedInitNode is used for rancherd where an exact machine (determined by labeling the
 // rkecontrolplane object) is desired to be the init node
-func (p *Planner) findAndDesignateFixedInitNode(rkeControlPlane *rkev1.RKEControlPlane, plan *plan.Plan) (bool, string, error) {
+func (p *Planner) findAndDesignateFixedInitNode(rkeControlPlane *rkev1.RKEControlPlane, plan *plan.Plan) (bool, string, *planEntry, error) {
```

Comment:
Unfortunately I wasn't able to implement this change, because `findInitNode` can return true with a `nil` init node to signify that there is an existing init node, with an empty join URL.